src/python/pants/engine/target.py

```diff
@@ -1647,6 +1651,25 @@ def unevaluated_transitive_excludes(self) -> UnparsedAddressInputs:
         )
 
 
+class TestTargetDependencies(Dependencies):
```

Comment:
This speaks to why I originally tried to move `conftest.py` into `python_library` rather than `python_tests`, which we rejected as too burdensome. It is indeed weird that tests are sometimes depended upon.

(Although even with that conftest change, it is still totally possible for one test to import another!)